container_main.py

```
@@ -66,19 +66,34 @@ def run_scheduler():
-
-        # Import and run the API server
-        from api_server import app
-
-        port = int(os.environ.get("PORT", 5000))
-        app.run(host="0.0.0.0", port=port, debug=False, use_reloader=False)
+        logger.info("Starting API server process via gunicorn")
+
+        # Strictly validate PORT to prevent command injection
+        port_str = os.environ.get("PORT", "5000")
+        if not port_str.isdigit():
+            logger.error(f"Invalid PORT value: {port_str}")
```

Ruff (0.15.0)

[warning] Logging statement uses f-string

(G004)